test/Models/WebMapTileServiceCatalogItemSpec.ts

```diff
@@ -0,0 +1,152 @@
+        wmts.getCapabilitiesUrl.indexOf(wmts.url || "undefined") === 0
+    ).toBeTruthy();
+  });
+
+  it("updates description from a GetCapabilities", async function() {
+    runInAction(() => {
+      wmts.setTrait("definition", "url", "test/WMTS/with_tilematrix.xml");
+      wmts.setTrait("definition", "layer", "Some_Layer1");
+    });
+
+    let description: String | undefined;
+    const cleanup = autorun(() => {
+      if (wmts.info !== undefined) {
+        const descSection = wmts.info.find(
+          section =>
+            section.name ===
+            i18next.t("models.webMapTileServiceCatalogItem.dataDescription")
+        );
+        if (
+          descSection !== undefined &&
+          descSection.content !== undefined &&
+          descSection.content !== null
+        ) {
+          description = descSection.content;
+        }
+      }
+    });
+
+    try {
+      await wmts.loadMetadata();
+      expect(description).toBe("description foo bar baz");
+    } finally {
+      cleanup();
+    }
+  });
+
+  it("correctly contstructs ImageryProvider with ResourceUrl", async function() {
+    runInAction(() => {
+      wmts.setTrait("definition", "url", "test/WMTS/with_tilematrix.xml");
+      wmts.setTrait("definition", "layer", "Some_Layer1");
+    });
+
+    let mapItems: ImageryParts[] = [];
+
+    const cleanup = autorun(() => {
+      mapItems = wmts.mapItems.slice();
+    });
+
+    try {
+      await wmts.loadMapItems();
+
+      expect(mapItems.length).toBe(1);
+      expect(mapItems[0].alpha).toBeCloseTo(0.8);
+      expect(
+        mapItems[0].imageryProvider instanceof WebMapTileServiceImageryProvider
+      ).toBeTruthy();
+      if (
+        mapItems[0].imageryProvider instanceof WebMapTileServiceImageryProvider
+      ) {
+        expect(mapItems[0].imageryProvider.url).toBe(
+          "https://some.provider/wmts/Some_Layer1/default/GoogleMapsCompatible_Level9/{TileMatrix}/{TileRow}/{TileCol}.png"
+        );
+      }
+    } finally {
+      cleanup();
+    }
+  });
+
+  it("correctly contstructs ImageryProvider", async function() {
+    runInAction(() => {
+      wmts.setTrait("definition", "url", "test/WMTS/with_tilematrix.xml");
+      wmts.setTrait("definition", "layer", "Some_Layer3");
+    });
+
+    let mapItems: ImageryParts[] = [];
+
+    const cleanup = autorun(() => {
+      mapItems = wmts.mapItems.slice();
+    });
+
+    try {
+      await wmts.loadMapItems();
+
+      expect(mapItems.length).toBe(1);
+      expect(mapItems[0].alpha).toBeCloseTo(0.8);
+      expect(
+        mapItems[0].imageryProvider instanceof WebMapTileServiceImageryProvider
+      ).toBeTruthy();
+      if (
+        mapItems[0].imageryProvider instanceof WebMapTileServiceImageryProvider
+      ) {
+        const url = new URI(mapItems[0].imageryProvider.url)
+          .search("")
+          .toString();
+        expect(url).toEqual("test/WMTS/with_tilematrix.xml");
+      }
+    } finally {
+      cleanup();
+    }
+  });
```

Comment:
These tests are failing due to an error occurring in Cesium here 

https://github.com/TerriaJS/cesium/blob/ecb3ede1797b0a2b79e3ee1dbd8bbff57da3edb4/Source/Scene/WebMapTileServiceImageryProvider.js#L320

```
mobx.module.js?7f83:1847 [mobx] Encountered an uncaught exception that was thrown by a reaction or observer component, in: 'Reaction[Autorun@1307]' TypeError: Cannot read property 'x' of undefined
    at new WebMapTileServiceImageryProvider (WebMapTileServiceImageryProvider.js?8e6a:302)
    at WebMapTileServiceCatalogItem.get (WebMapTileServiceCatalogItem.ts?1b33:1)
    at trackDerivedFunction (mobx.module.js?7f83:786)
    at ComputedValue.computeValue (mobx.module.js?7f83:1271)
    at ComputedValue.trackAndCompute (mobx.module.js?7f83:1256)
    at ComputedValue.get (mobx.module.js?7f83:1216)
    at shouldCompute (mobx.module.js?7f83:706)
    at ComputedValue.get (mobx.module.js?7f83:1215)
    at shouldCompute (mobx.module.js?7f83:706)
    at Reaction.runReaction (mobx.module.js?7f83:1787)
```

Do you have any idea why this is happening? I'm not sure

Reply:
Will run it later tonight and provide a feedback, but this is something with tiles, probably thrown in this line https://github.com/CesiumGS/cesium/blob/1.73/Source/Scene/WebMapTileServiceImageryProvider.js#L299 (only place cesium is looking for .x)

Reply:
In L320 Cesium calls function `positionToTileXY` which then checks if the supplied position is falling in the maximum rectangle defined for that scheme ([this func](https://github.com/TerriaJS/cesium/blob/ecb3ede1797b0a2b79e3ee1dbd8bbff57da3edb4/Source/Scene/WebMapTileServiceImageryProvider.js#L320)). 
The rectangle defined for the scheme has the following bounds
`{west: -3.141592653589793, south: -1.4844222297453324, east: 3.141592653589793, north: 1.4844222297453322}`
while we send 
`{west: -3.141592653589793, south: -1.4844222335884323, east: 3.141592653589793, north: 1.4844222335884323}`
here is north coordinate bigger than the coordinate defined for tiling scheme (last decimal).

There are two solutions that come to my mind to create TillingScheme with the specific rectangle size, or to reduce the size of the rectangle if it is falling outside tillingScheme rectangle.

Reply:
I have added a simple check here https://github.com/TerriaJS/cesium/pull/78/files#diff-95a7930932cbafb5bb06b2cae7637aa4R319-R326
I will revisit this later